src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1217,6 +1218,41 @@ public override int GetHashCode()
             return (this.Name.GetHashCode());
         }
 
+        internal static unsafe int GetIgnoreCaseHash(string source)
+        {
+            // Do not allocate on the stack if string is empty
+            if (source.Length == 0)
+            {
+                return source.GetHashCode();
+            }
+
+            Span<char> span = source.Length < 256 ? 
+                stackalloc char[source.Length] : 
```

Comment:
It is a bit efficient to just allocate fixed size stack buffer. The variable size just adds a bit of extra overhead.

Reply:
right, I should have thought about that. Thanks.